src/LibLog/LibLog.cs

```diff
@@ -351,11 +368,44 @@ public interface ILogProvider
     }
 
     /// <summary>
+    /// Static class that allows you to change the log provider used by LibLog. This class
+    /// is public static on purpose, because it's meant to be used by client applications. 
+    /// HOwever, the day to day methods called by consumers of liblog (such as LogProvider.GetLogger())
+    /// are by default ONLY accessable to the component itself. 
+    /// </summary>
+    public static class LogProviderConfiguration
+    {
+        private static dynamic _currentLogProvider;
+        /// <summary>
+        /// Sets the current log provider.
+        /// </summary>
+        /// <param name="logProvider">The log provider.</param>
+        public static void SetCurrentLogProvider(dynamic logProvider)
```

Comment:
I think this is going to give us some interesting runtime problems. There are also different versions of LibLog with different ILogProvider (and different ILog) signatures. How will a 3rd party know what to put here?